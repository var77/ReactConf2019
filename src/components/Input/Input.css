/* ====================================================
Recreating the email field from https://webflow.com/cms. Just an experiment - not as cross-browser friendly as the original.
Changed:
- animated gradient bar to :after element
- flexbox for layout
==================================================== */
html {
  box-sizing: border-box;
  font-size: 10px;
}

*, *:before, *:after {
  box-sizing: inherit;
}

body, ul, li {
  margin: 0;
  padding: 0;
  background-image: -webkit-radial-gradient(0% 0%, circle, #373b52, #252736 250%, #1d1e26);
  background-image: radial-gradient(circle at 0% 0%, #373b52, #252736 250%, #1d1e26);
}

li {
  list-style: none;
}

p, h1, h2, h3, h4, h5, h6 {
  margin-top: 0;
}

a {
  text-decoration: none;
}

input {
  border-style: none;
  background: transparent;
  outline: none;
}

button {
  padding: 0;
  background: none;
  border: none;
  outline: none;
}

body {
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  /* background-image: -webkit-radial-gradient(0% 0%, circle, #373b52, #252736 51%, #1d1e26);
  background-image: radial-gradient(circle at 0% 0%, #373b52, #252736 51%, #1d1e26); */
}

h1.demo {
  text-align: center;
  font-size: 2.4rem;
  font-weight: normal;
  margin-bottom: 1rem;
  color: #f5f6ff;
}

a.demo {
  text-align: center;
  font-size: 1.6rem;
  font-weight: normal;
  color: rgba(202, 205, 239, 0.8);
  margin-bottom: 3rem;
}

.demo-flex-spacer {
  -webkit-box-flex: 1;
      -ms-flex-positive: 1;
          flex-grow: 1;
}

.container {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  height: 100vh;
  max-width: 1600px;
  padding: 0 15px;
  margin: 0 auto;
}

@-webkit-keyframes gradient {
  0% {
    background-position: 0 0;
  }
  100% {
    background-position: 100% 0;
  }
}

@keyframes gradient {
  0% {
    background-position: 0 0;
  }
  100% {
    background-position: 100% 0;
  }
}
.webflow-style-input {
  position: relative;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-direction: row;
          flex-direction: row;
  width: 100%;
  max-width: 400px;
  margin: 0 auto;
  border-radius: 2px;
  padding: 1.4rem 2rem 1.6rem;
  background: rgba(57, 63, 84, 0.8);
}
.webflow-style-input:after {
  content: "";
  position: absolute;
  left: 0px;
  right: 0px;
  bottom: 0px;
  z-index: 999;
  height: 2px;
  border-bottom-left-radius: 2px;
  border-bottom-right-radius: 2px;
  background-position: 0% 0%;
  background: -webkit-linear-gradient(left, #B294FF, #57E6E6, #FEFFB8, #57E6E6, #B294FF, #57E6E6);
  background: linear-gradient(to right, #B294FF, #57E6E6, #FEFFB8, #57E6E6, #B294FF, #57E6E6);
  background-size: 500% auto;
  -webkit-animation: gradient 3s linear infinite;
          animation: gradient 3s linear infinite;
}

.webflow-style-input input {
  -webkit-box-flex: 1;
      -ms-flex-positive: 1;
          flex-grow: 1;
  color: #BFD2FF;
  font-size: 1.8rem;
  line-height: 2.4rem;
  vertical-align: middle;
}
.webflow-style-input input::-webkit-input-placeholder {
  color: #7881A1;
}

.webflow-style-input button {
  color: #7881A1;
  font-size: 2.4rem;
  line-height: 2.4rem;
  vertical-align: middle;
  -webkit-transition: color .25s;
  transition: color .25s;
}
.webflow-style-input button:hover {
  color: #BFD2FF;
}


.wrong-webflow-style-input {
  position: relative;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-direction: row;
          flex-direction: row;
  width: 100%;
  max-width: 400px;
  margin: 0 auto;
  border-radius: 2px;
  padding: 1.4rem 2rem 1.6rem;
  background: rgba(57, 63, 84, 0.8);
}
.wrong-webflow-style-input:after {
  content: "";
  position: absolute;
  left: 0px;
  right: 0px;
  bottom: 0px;
  z-index: 999;
  height: 2px;
  border-bottom-left-radius: 2px;
  border-bottom-right-radius: 2px;
  background-position: 0% 0%;
  background: -webkit-linear-gradient(left, red, red, red, red, red, red);
  background: linear-gradient(to right, red, red, red, red, red, red);
  background-size: 500% auto;
  -webkit-animation: gradient 3s linear infinite;
          animation: gradient 3s linear infinite;
}

.wrong-webflow-style-input input {
  -webkit-box-flex: 1;
      -ms-flex-positive: 1;
          flex-grow: 1;
  color: #ff3e17;
  font-size: 1.8rem;
  line-height: 2.4rem;
  vertical-align: middle;
}
.wrong-webflow-style-input input::-webkit-input-placeholder {
  color: #7881A1;
}

.wrong-webflow-style-input button {
  color: #7881A1;
  font-size: 2.4rem;
  line-height: 2.4rem;
  vertical-align: middle;
  -webkit-transition: color .25s;
  transition: color .25s;
}
.wrong-webflow-style-input button:hover {
  color: #BFD2FF;
}

.card {
  background: #fff;
  border-radius: 2px;
  display: flex;
  height: 100%;
  margin: 1rem;
  position: relative;
  width: 100%;
  justify-content: space-evenly;
  align-items: center;
  flex-direction: column;
  background-image: -webkit-radial-gradient(0% 0%, circle, #373b52, #252736 100%, #1d1e26);
  background-image: radial-gradient(circle at 0% 0%, #373b52, #252736 100%, #1d1e26);
}

.card-1 {
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  transition: all 0.3s cubic-bezier(.25,.8,.25,1);
}

.card-1:hover {
  box-shadow: 0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22);
}

.card-2 {
  box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
}

.card-3 {
  box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23);
}

.card-4 {
  box-shadow: 0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22);
}

.card-5 {
  box-shadow: 0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22);
}

.headerText{
    margin: 10px;
    font-size: 54px;
    color: white;
}

.mainCard{
    width: 46vw;
    height: 46vh;
    display: 'flex';
    flex-direction: 'column';
}

.inputText {
    color: white;
    font-size: 18px;
    text-align: left;
}

.checkbox-input {
    display: flex;
    flex-direction: row;
    align-items: center;
}

.inputSize {
    width: 45%;
}

.checkboxStyle {
  transform: scale(1.5);
  margin-left: 10px;
}