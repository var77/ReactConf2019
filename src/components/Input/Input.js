import React, { Component, useState, useEffect } from 'react';
import './Input.css';

const validateEmail = (val) => {
    const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
    return re.test(String(val).toLowerCase());
};

const Input = ({ value, onChange, disabled, email, ...props }) => {

    const [isValidEmail, setEmailValid] = useState(email ? validateEmail(value) : true);

    useEffect(() => {
        email && setEmailValid(validateEmail(value));
    }, [value]);

    return (
        <div className={(isValidEmail ? 'webflow-style-input' : 'wrong-webflow-style-input')}>
            <input type="text" {...props} value={value} onChange={onChange} disabled={disabled}/>
        </div>
    );
};

export default Input;