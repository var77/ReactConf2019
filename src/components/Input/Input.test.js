import React from 'react';
import { shallow } from 'enzyme';

import Input from './Input';

describe('Input Component', () => {
    it ('should render without crashing', () => {
        const el = shallow(<Input />);
        expect(el).toMatchSnapshot();
    });

    it ('should check the checkbox', () => {
        let val = 'initial value';
        const changedVal = 'value changed';
        let el = null;
        const onChange = jest.fn((e) => el.setProps({ value: e.target.value }));
        el = shallow(<Input onChange={onChange} value={val} />);
        expect(el.props().value).toBe(val);
        el.simulate('change', { target: { value: changedVal } });
        expect(el.props().value).toBe(changedVal);
    });
});