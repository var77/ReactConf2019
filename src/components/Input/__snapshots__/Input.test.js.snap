// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render without crashing 1`] = `
<div
  className="webflow-style-input"
>
  <input
    type="text"
  />
</div>
`;
